From bda2674f22b58bd32802b2057a05efada6155bae Mon Sep 17 00:00:00 2001
From: Scott James Remnant <scott@netsplit.com>
Date: Fri, 20 May 2011 14:06:29 -0700
Subject: [PATCH] configure: allow usb.ids location to be specified

We already allow the pci.ids location to be specified, so add a
patch doing the same for usb.ids. Please don't make me explain
why this is necessary, it will only make you cry.

Signed-off-by: Scott James Remnant <scott@netsplit.com>
Signed-off-by: Kay Sievers <kay.sievers@vrfy.org>
Acked-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 configure.ac |   14 ++++++++++++--
 1 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3646f93..cfdb3bf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -93,8 +93,18 @@ if test "x$enable_hwdb" = xyes; then
 		AC_CHECK_FILES([/usr/share/misc/pci.ids], [pciids=/usr/share/misc/pci.ids])
 	fi
 
-	PKG_CHECK_MODULES(USBUTILS, usbutils >= 0.82)
-	AC_SUBST([USB_DATABASE], [$($PKG_CONFIG --variable=usbids usbutils)])
+	AC_ARG_WITH(usb-ids-path,
+		[AS_HELP_STRING([--with-usb-ids-path=DIR], [Path to usb.ids file])],
+		[USB_DATABASE=${withval}],
+		[if test -n "$usbids" ; then
+			USB_DATABASE="$usbids"
+		else
+			PKG_CHECK_MODULES(USBUTILS, usbutils >= 0.82)
+			AC_SUBST([USB_DATABASE], [$($PKG_CONFIG --variable=usbids usbutils)])
+		fi])
+	AC_MSG_CHECKING([for USB database location])
+	AC_MSG_RESULT([$USB_DATABASE])
+	AC_SUBST(USB_DATABASE)
 
 	AC_ARG_WITH(pci-ids-path,
 		[AS_HELP_STRING([--with-pci-ids-path=DIR], [Path to pci.ids file])],
-- 
1.7.3.1

